### ISSUE TITLE
missing entity resolver in BlogCategoryRepository::__contruct

### ISSUE THREAD

**Issue by vitek-rostislav** [shopsys/shopsys#3066]
## Describe the bug
We need to use entityNameResolver [here](https://github.com/shopsys/shopsys/blob/15.0/packages/framework/src/Model/Blog/Category/BlogCategoryRepository.php#L24), otherwise it fails on an error when `BlogCategory` is extended on a project

```
[Gedmo\Exception\UnexpectedValueException]
  Tree object class: Shopsys\FrameworkBundle\Model\Blog\Category\BlogCategory must have tree metadata at this point

Exception trace:
  at /var/www/html/vendor/gedmo/doctrine-extensions/src/Tree/TreeListener.php:119
 Gedmo\Tree\TreeListener->getStrategy() at /var/www/html/vendor/gedmo/doctrine-extensions/src/Tree/Entity/Repository/NestedTreeRepository.php:1185
 Gedmo\Tree\Entity\Repository\NestedTreeRepository->validate() at /var/www/html/vendor/gedmo/doctrine-extensions/src/Tree/Entity/Repository/AbstractTreeRepository.php:68
 Gedmo\Tree\Entity\Repository\AbstractTreeRepository->__construct() at /var/www/html/vendor/shopsys/framework/src/Model/Blog/Category/BlogCategoryRepository.php:29
 Shopsys\FrameworkBundle\Model\Blog\Category\BlogCategoryRepository->__construct() at /var/www/html/var/cache/dev/ContainerTtJj7fn/getBlogCategoryRepositoryService.php:29
 ContainerTtJj7fn\getBlogCategoryRepositoryService::do() at /var/www/html/var/cache/dev/ContainerTtJj7fn/App_KernelDevDebugContainer.php:1482
 ContainerTtJj7fn\App_KernelDevDebugContainer->load() at /var/www/html/var/cache/dev/ContainerTtJj7fn/getBlogCategoryFacadeService.php:40
 ContainerTtJj7fn\getBlogCategoryFacadeService::do() at /var/www/html/var/cache/dev/ContainerTtJj7fn/App_KernelDevDebugContainer.php:1482
 ContainerTtJj7fn\App_KernelDevDebugContainer->load() at /var/www/html/var/cache/dev/ContainerTtJj7fn/getBlogArticleFacadeService.php:42
 ContainerTtJj7fn\getBlogArticleFacadeService::do() at /var/www/html/var/cache/dev/ContainerTtJj7fn/App_KernelDevDebugContainer.php:1482
 ContainerTtJj7fn\App_KernelDevDebugContainer->load() at /var/www/html/var/cache/dev/ContainerTtJj7fn/getDoctrine_FixturesLoadCommandService.php:185
 ContainerTtJj7fn\getDoctrine_FixturesLoadCommandService::do() at /var/www/html/var/cache/dev/ContainerTtJj7fn/App_KernelDevDebugContainer.php:1482
 ContainerTtJj7fn\App_KernelDevDebugContainer->load() at /var/www/html/vendor/symfony/dependency-injection/Container.php:427
 Symfony\Component\DependencyInjection\Container->getService() at /var/www/html/vendor/symfony/dependency-injection/Argument/ServiceLocator.php:42
 Symfony\Component\DependencyInjection\Argument\ServiceLocator->get() at /var/www/html/vendor/symfony/console/CommandLoader/ContainerCommandLoader.php:45
 Symfony\Component\Console\CommandLoader\ContainerCommandLoader->get() at /var/www/html/vendor/symfony/console/Application.php:596
 Symfony\Component\Console\Application->has() at /var/www/html/vendor/symfony/console/Application.php:685
 Symfony\Component\Console\Application->find() at /var/www/html/vendor/symfony/framework-bundle/Console/Application.php:116
 Symfony\Bundle\FrameworkBundle\Console\Application->find() at /var/www/html/vendor/symfony/console/Application.php:259
 Symfony\Component\Console\Application->doRun() at /var/www/html/vendor/symfony/framework-bundle/Console/Application.php:82
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /var/www/html/vendor/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at /var/www/html/vendor/symfony/runtime/Runner/Symfony/ConsoleApplicationRunner.php:54
 Symfony\Component\Runtime\Runner\Symfony\ConsoleApplicationRunner->run() at /var/www/html/vendor/autoload_runtime.php:35
 require() at /var/www/html/app/autoload.php:18
 require_once() at /var/www/html/bin/console:9
```

Or, and that would be probably better, we could override `getClassMetadata` method in our `EntityManagerDecorator` and use the resolver there

**Comment by grossmannmartin** [comment]
I believe this may be closed, @vitek-rostislav?

**Comment by vitek-rostislav** [comment]
@grossmannmartin Yes, you are right, it was set to be closed by #3068, but the PR was merged into `14.0` which is not a main branch anymore, so the automated issue closing probably did not take the merge into account :thinking:
